lib/promscrape/discovery/kubernetes/api_watcher.go

```diff
@@ -458,6 +464,11 @@ type urlWatcher struct {
 	apiURL    string
 	gw        *groupWatcher
 
+	refCount int
+	wg       sync.WaitGroup
```

Comment:
Do we actually need a wait group for that? Cancelling the context is the best thing we can do here, so why should we wait for all gorotuines to exit? Why we need this synchronization?

Reply:
Thought about this one more time and removed it, here is the reason why we should not use WG here:
> This is unnecessary since context will is cancelled and new requests will not be sent. Also, using waitgroup will increase time required to perform reload which might result in missed scrapes.